xrpl/websocket/helpers.go

```diff
@@ -0,0 +1,50 @@
+package websocket
+
+import (
+	"fmt"
+
+	binarycodec "github.com/Peersyst/xrpl-go/binary-codec"
+	transaction "github.com/Peersyst/xrpl-go/xrpl/transaction"
+	"github.com/Peersyst/xrpl-go/xrpl/wallet"
+)
+
+// getSignedTx returns a fully signed transaction blob. It checks whether the
+// transaction already contains a signature or public key and, if not, optionally
+// autofills missing fields and signs the transaction using the provided wallet.
+// It verifies the resulting blob includes a signature, returning an error if absent.
+func getSignedTx(client *Client, tx transaction.FlatTransaction, autofill bool, wallet *wallet.Wallet) (string, error) {
```

Comment:
This function should check the following behavior:

1. Check if the tx is signed (`TxSignature` and `SigningPubKey` fields are both not empty.
2. If signed encode the transaction and submit it, then return.
3. If not signed, check if wallet is nil.
4. If wallet is nil, throw error.
5. If not, check if autofill option is true. If true, autofill. Then sign and return